### PR TITLE
[forge] label test runner pods with image tag

### PR DESCRIPTION
**Pull request by rustielin**
### Description

Make forge test runner pod names more descriptive and non-overlapping. Include in the pod name the Forge namespace, timestamp, and image_tag/git_ref. 

Also label all resources managed by Forge (the test runner, all resources in the included helm releases) with labels `forge-namespace` and `forge-image-tag`. We could consider labeling with the Github actions run ID too, but we'd need to propagate that into the rust code.

Also fix a big that was re-introduced where the test runner acquires a namespace by pre-empting another test runner, but fails on the k8s API 409 error code instead of continuing.

### Test Plan

Run forge with k8s test runner. Then run another one in the same namespace and let them pre-empt each other.

Run it the first time with a known namespace. Test runner pod starts with the name `forge-rustie-test-1659115106-e9dcb772a90b6941d73053bbc473257c768`

```
$ IMAGE_TAG=$(git rev-parse HEAD) FORGE_NAMESPACE=forge-rustie-test ./testsuite/run_forge.sh
```

Run it a second time and preempt. Test runner pod starts with the name `forge-rustie-test-1659115193-e9dcb772a90b6941d73053bbc473257c768`
```
$ IMAGE_TAG=$(git rev-parse HEAD) FORGE_NAMESPACE=forge-rustie-test ./testsuite/run_forge.sh
...
{"level":"INFO","source":{"package":"forge","file":"testsuite/forge/src/backend/k8s/cluster_helper.rs:701"},"thread_name":"main","hostname":"forge-rustie-test-1659115193-e9dcb772a90b6941d73053bbc473257c76","timestamp":"2022-07-29T17:19:57.033416Z","message":"Namespace forge-rustie-test already exists, continuing with it"}

test continues as it should...
```

The first instance exits with a good exit code, but knows that it has been preempted. This will help keep the `main` branch green when many PRs land in succession and their Forge runs are pre-empted, eliminating some of the noise we've seen on `main` with preempted Forge runs reporting "failed"
```
Forge test preempted on `e9dcb772a90b6941d73053bbc473257c768cffe7`: command not found
...
Forge exit with: 0
```

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2276)

### CODE REVIEW

File: testsuite/forge/src/backend/k8s/cluster_helper.rs [hunk 1]
```diff
@@ -394,6 +395,12 @@ pub async fn uninstall_testnet_resources(kube_namespace: String) -> Result<()> {
     Ok(())
 }
 
+fn generate_new_era() -> String {
+    let mut rng = rand::thread_rng();
+    let r: u8 = rng.gen();
+    format!("forge{}", r)
```
Comment (sitalkedia) [inline testsuite/forge/src/backend/k8s/cluster_helper.rs:401]: Doesn't the era need to be an integer?

Comment (rustielin) [reply testsuite/forge/src/backend/k8s/cluster_helper.rs:401]: nope. under the hood, it's just used to name kubernetes resources, so it's any string

**Comment by sherry-x** [comment]
General feedback: whenever making changes that touch single node deployment / testnet deployment, could you please make sure they're tested as well? Ideally we should have a cluster of each dedicated to CICD testing, and have GHA to deploy them

**Review by sitalkedia — APPROVED** [review]
LGTM

**Comment by rustielin** [comment]
>  whenever making changes that touch single node deployment / testnet deployment, could you please make sure they're tested as well?

Yea we should have a separate pipeline to test the full deployment in a GHA. Just don't have the bandwidth for it atm. It's our first issue on linear 😄 : https://linear.app/aptoslabs/issue/PE-1/test-helm-charts

Generally, Forge / nightly Forge can catch these somewhat, but ofc It's not capturing all the cases where new features are on or off.

**Pull request merged**